braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -450,18 +450,32 @@ private static Date dateFromString(String date) {
         }
     }
 
-    // compare two address objects and return accordingly
+    // compare two address objects and return false if there is a change in address or true otherwise
     private static boolean compareAddress(@Nullable RudderTraits.Address curr, @Nullable RudderTraits.Address prev) {
-        if (prev == null && curr != null) {
-            return true;
+        if (prev == null & curr != null) {
+            return false;
         }
+        if (prev != null && curr != null) {
+            // if previous identify call doesn't contain city but current one do
+            if (prev.getCity() == null && curr.getCity() != null) {
+                return false;
+            }
+            if (prev.getCity() != null && curr.getCity() != null)
+                if (!(prev.getCity().equals(curr.getCity()))) {
```

Comment:
Having an individual `return` statement, like thatm would cause immediate return from the method, which is not what we wanted. We need to return only when that condition satisfies.
Please LMK what would be better